### ISSUE TITLE
Support file_header_template

### ISSUE THREAD

**Issue by RedX2501** [DotNetAnalyzers/StyleCopAnalyzers#3605]
It seems to have problems with multi-line headers.

When you have something like:

```
file_header_template = this\nspans\nmultiple\nlines
```

the warning will be that it expects it to be all in one line.

**Comment by bjornhellander** [comment]
Which warning do you see?
And which StyleCop Analyzers version are you using?

**Comment by RedX2501** [comment]
StyleCop.Analyzers 1.2.0-beta.354.

Based on your question I checked and there is the new 435. I updated to this version and still get that both differ.


stylecop.json contents:

```json
{
  // ACTION REQUIRED: This file was automatically added to your project, but it
  // will not take effect until additional steps are taken to enable it. See the
  // following page for additional information:
  //
  // https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/EnableConfiguration.md

  "$schema": "https://raw.githubusercontent.com/DotNetAnalyzers/StyleCopAnalyzers/master/StyleCop.Analyzers/StyleCop.Analyzers/Settings/stylecop.schema.json",
  "settings": {
  }
}

```

editorconfig
```

# All files
[*]
indent_style = space

# Xml files
[*.xml]
indent_size = 2

[*.cs]
file_header_template =  <copyright file="{fileName}" company="My company">multiple\nlines\nwith {fileName}\n</copyright>
```

Warning:
```
Warning	SA1636	The file header copyright text should match the copyright text from the settings.	The file header at the top of a C# code file does not contain the appropriate copyright text.
```

Header in source file:

```c#
// <copyright file="ModelKitModel.cs" company="My company">multiple
// lines
// with ModelKitModel.cs
// </copyright>

```

**Comment by sharwell** [comment]
It looks like we don't do any processing of `\n` or `\r\n` here:

https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/feea263906c6ac13b832dbde2c2d15901795b6fb/StyleCop.Analyzers/StyleCop.Analyzers/Settings/ObjectModel/DocumentationSettings.cs#L248-L249

**Comment by RedX2501** [comment]
@sharwell  I've had a quick look at the code and I didn't find also any handling for the string that comes from the JSON. Is it safe to assume that it's because the JSON deserializer is actually handing the line wrapping?

If that's the case would it be enough at that line to handle the wrapping for the value coming from the editorconfig or do I need to investigate more?

**Comment by bjornhellander** [comment]
@RedX2501 JSON already interprets `\n` as a line feed character, so only the code that handles the .editorconfig setting needs to handle escape sequences.
